### PR TITLE
[SPARK-49801][PYTHON][PS][BUILD] Update `pandas` to 2.2.3

### PR DESCRIPTION
**Pull request by bjornjorgensen**
### What changes were proposed in this pull request?
Update pandas from 2.2.2 to 2.2.3

### Why are the changes needed?
[Release notes](https://pandas.pydata.org/pandas-docs/version/2.2.3/whatsnew/v2.2.3.html)

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @bjornjorgensen .

**Review by HyukjinKwon — APPROVED** [review]
cc @itholic

**Review by xinrong-meng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: python/pyspark/pandas/supported_api_gen.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 MAX_MISSING_PARAMS_SIZE = 5
 COMMON_PARAMETER_SET = {"kwargs", "args", "cls"}
 MODULE_GROUP_MATCH = [(pd, ps), (pdw, psw), (pdg, psg)]
-PANDAS_LATEST_VERSION = "2.2.2"
+PANDAS_LATEST_VERSION = "2.2.3"
```
Comment (dongjoon-hyun) [inline python/pyspark/pandas/supported_api_gen.py:41]: Oh, this seems to break `GitHub Action Jekyll`.

- https://github.com/apache/spark/actions/runs/11063509911/job/30742286270

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/sphinx/config.py", line 332, in eval_config_file
    exec(code, namespace)
  File "/home/runner/work/spark/spark/python/docs/source/conf.py", line 33, in <module>
    generate_supported_api(output_rst_file_path)
  File "/home/runner/work/spark/spark/python/pyspark/pandas/supported_api_gen.py", line 102, in generate_supported_api
    _check_pandas_version()
  File "/home/runner/work/spark/spark/python/pyspark/pandas/supported_api_gen.py", line 116, in _check_pandas_version
    raise ImportError(msg)
ImportError: Warning: pandas 2.2.3 is required; your version is 2.2.2
```

Could you take a look at this failure, @bjornjorgensen ?

Also, cc @yaooqinn , too.

Comment (yaooqinn) [reply python/pyspark/pandas/supported_api_gen.py:41]: Yea, we need to update that yaml file too

**Pull request closed without merging**